sycl/test/on-device/extensions/matrix-amx-bf16-test.cpp

```diff
@@ -0,0 +1,186 @@
+// RUN: %clangxx -march=sapphirerapids -fsycl -O2 %s -o %t.out
+#include <CL/sycl.hpp>
+#include <array>
+#include <assert.h>
+#include <cstdint>
+#include <cstdio>
+#include <immintrin.h>
```

Comment:
Does the user need to add all these include for such simple code, please double check?
can we move #include <immintrin.h> to matrix-amx.hpp to avoid the user adding it here?